earth_enterprise/src/fusion/portableglobe/geglxinfo.cpp

```diff
@@ -66,65 +73,91 @@ void GetXY(int level, uint64 btree, uint64* x, uint64* y) {
+  if (skipSmall && buffer.size() <= 153) {
+    std::cout << "Skipping small: " << path << std::endl;
+  } else {
+    khEnsureParentDir(path);
+    khWriteSimpleFile(path, &buffer[0], buffer.size());
+  }
+}
+
+bool getPackageFileLocs(GlcUnpacker* const unpacker,
+                    const std::string& index_file,
+                    const std::string& data_file,
                     bool is_composite,
                     int layer_idx,
-                    uint64 start_idx,
-                    uint64 end_idx) {
-  PackageFileLoc index_file_loc;
-  std::string index_file = "mapdata/index";
-  PackageFileLoc data_file_loc;
-  std::string data_file = "mapdata/pbundle_0000";
+                    PackageFileLoc& index_file_loc,
+                    PackageFileLoc& data_file_loc) {
   // For glcs, we need to look inside each layer for the indexes.
   if (is_composite) {
-    // Determine if the glc is 2d, 3d, or both
-    for (int i = 0; i < unpacker->IndexSize(); ++i) {
-      std::string package_file = unpacker->IndexFile(i);
-      std::string layer_suffix = package_file.substr(package_file.size() - 3);
-      if (layer_suffix == "glb") {
-        std::cout << "Only 2d is supported." << std::endl;
-        return;
-      } else if (layer_suffix == "glm") {
-        std::cout << "2d glc" << std::endl;
-      }
-    }
-
     int layer_index = unpacker->LayerIndex(layer_idx);
     if (!unpacker->FindLayerFile(
            index_file.c_str(), layer_index, &index_file_loc)) {
        std::cout << "Unable to find layer index: " << index_file << std::endl;
-       return;
+       return false;
     }
 
     if (!unpacker->FindLayerFile(
            data_file.c_str(), layer_index, &data_file_loc)) {
        std::cout << "Unable to find layer data: " << data_file << std::endl;
-       return;
+       return false;
     }
 
   // For glms and glbs, just use the corresponding data index and divide
   // by its size (sizeof(IndexItem)).
   } else {
     if (!unpacker->FindFile(index_file.c_str(), &index_file_loc)) {
         std::cout << "Unable to find index: " << index_file << std::endl;
-      return;
+      return false;
     }
     if (!unpacker->FindFile(data_file.c_str(), &data_file_loc)) {
         std::cout << "Unable to find data: " << data_file << std::endl;
-      return;
+      return false;
     }
   }
 
+  return true;
+}
+
+void extractAllPackets(GlcUnpacker* const unpacker,
+                    PortableGlcReader* const reader,
+                    uint64 start_idx,
+                    uint64 end_idx,
+                    int layer_idx,
+                    bool is_composite,
+                    const std::string& index_file,
+                    const std::string& data_file) {
+  PackageFileLoc index_file_loc;
+  PackageFileLoc data_file_loc;
+
+  if (!getPackageFileLocs(unpacker, index_file, data_file, is_composite, layer_idx,
+                          index_file_loc, data_file_loc)) {
+    return;
```

Comment:
Should we output something here?

Reply:
I didn't output anything additional here because in getPackageFileLocs it prints a more specific message on the failure before returning false.